pkg/odo/cli/component/list.go

```diff
@@ -104,57 +144,118 @@ func (lo *ListOptions) Run() (err error) {
 		}
 		return nil
 	}
-	var components component.ComponentList
 
-	if lo.allAppsFlag {
-		// retrieve list of application
-		apps, err := application.List(lo.Client)
+	// non --path workflow below
+	// read the code like
+	// -> experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+	// -> non-experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+
+	// experimental workflow
+
+	if experimental.IsExperimentalModeEnabled() {
+
+		var dpl *appsv1.DeploymentList
+		var err error
+
+		// TODO: wrap this into a component list for docker support
+		if lo.allAppsFlag {
+			dpl, err = lo.KClient.ListAllDeployments()
+
+		} else {
+			dpl, err = lo.KClient.ListDeployments(lo.Application)
+		}
+
 		if err != nil {
 			return err
 		}
 
-		var componentList []component.Component
+		if len(dpl.Items) != 0 {
+			lo.hasDevfileComponents = true
+			w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+			fmt.Fprintln(w, "Devfile Components: ")
+			fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "PROJECT", "\t", "TYPE", "\t", "REVISION")
+			for _, comp := range dpl.Items {
+				app := comp.Labels[applabels.ApplicationLabel]
+				cmpType := comp.Labels[componentlabels.ComponentTypeLabel]
+				revision := comp.Annotations["deployment.kubernetes.io/revision"]
```

Comment:
do we need to display the revision?

Reply:
It allowed me to understand is my application was updated or not, does it reduce the UX?

Reply:
I seems s2i components don't have this column. Can we have the component `STATE` instead?

Reply:
added